test/test_with_xpress.jl

```diff
@@ -1440,3 +1440,138 @@ end
+#         "city": "LosAngeles"
+#     },
+#     "ElectricStorage": {
+#         "total_rebate_per_kw": 100.0,
+#         "macrs_option_years": 5,
+#         "can_grid_charge": true,
+#         "macrs_bonus_pct": 0.4,
+#         "macrs_itc_reduction": 0.5,
+#         "total_itc_pct": 0,
+#         "installed_cost_per_kw": 1000.0,
+#         "installed_cost_per_kwh": 500.0,
+#         "replace_cost_per_kw": 460.0,
+#         "replace_cost_per_kwh": 230.0
+#     },
+#     "ElectricTariff": {
+#         "urdb_label": "5ed6c1a15457a3367add15ae"
+#     },
+#     "Financial": {
+#         "escalation_pct": 0.026,
+#         "offtaker_discount_pct": 0.081,
+#         "owner_discount_pct": 0.081,
+#         "analysis_years": 20,
+#         "offtaker_tax_pct": 0.4,
+#         "owner_tax_pct": 0.4,
+#         "om_cost_escalation_pct": 0.025
+#     }
+# }}}
+
+@testset "Custom REopt logger" begin
+    
+    # Create minimal test case
+    req = Dict()
+    req["Site"] = Dict(
+        "longitude" => -155.8394336,
+        "latitude" => 65.0
+    )
+    req["ElectricLoad"] = Dict(
+        "doe_reference_name" => "MidriseApartment",
+        "annual_kwh" => 100000.0
+    )
+    req["ElectricTariff"] = Dict(
+        "urdb_label" => "5ed6c1a15457a3367add15ae"
+    )
+
+    # No Errors, only warnings
+    m = Model(optimizer_with_attributes(Xpress.Optimizer, "OUTPUTLOG" => 0))
+    r = run_reopt(m,req)
+    @test r["status"] == "optimal"
+    @test "Messages" ∈ keys(r)
+    @test "errors" ∈ keys(r["Messages"])
+    @test "warnings" ∈ keys(r["Messages"])
+
+    # No Errors, only warnings
+    m = Model(optimizer_with_attributes(Xpress.Optimizer, "OUTPUTLOG" => 0))
+    r = run_reopt((m,REoptInputs(req)))
+    @test r["status"] == "optimal"
+    @test "Messages" ∈ keys(r)
+    @test "errors" ∈ keys(r["Messages"])
+    @test "warnings" ∈ keys(r["Messages"])
+
+    # No Errors, only warnings
+    m1 = Model(optimizer_with_attributes(Xpress.Optimizer, "OUTPUTLOG" => 0))
+    m2 = Model(optimizer_with_attributes(Xpress.Optimizer, "OUTPUTLOG" => 0))
+    r = run_reopt([m1,m2],req)
+    @test r["status"] == "optimal"
+    @test "Messages" ∈ keys(r)
+    @test "errors" ∈ keys(r["Messages"])
+    @test "warnings" ∈ keys(r["Messages"])
+
+    # No Errors, only warnings
+    m = Model(optimizer_with_attributes(Xpress.Optimizer, "OUTPUTLOG" => 0))
+    r = run_reopt(m,"./scenarios/pv.json")
+    @test r["status"] == "optimal"
+    @test "Messages" ∈ keys(r)
+    @test "errors" ∈ keys(r["Messages"])
+    @test "warnings" ∈ keys(r["Messages"])
+
+    # No Errors, only warnings
+    m1 = Model(optimizer_with_attributes(Xpress.Optimizer, "OUTPUTLOG" => 0))
+    m2 = Model(optimizer_with_attributes(Xpress.Optimizer, "OUTPUTLOG" => 0))
+    r = run_reopt([m1,m2],"./scenarios/pv.json")
+    @test r["status"] == "optimal"
+    @test "Messages" ∈ keys(r)
+    @test "errors" ∈ keys(r["Messages"])
+    @test "warnings" ∈ keys(r["Messages"])
+
+    # No Errors, no warnings
```

Comment:
That test is checking that the logging functionality is working correctly when calling the `REoptInputs` function directly, as the API v3 does to get emission defaults. Since we expect to get type REoptInputs back we just check for that. If we get a dictionary instead then there was an error in REoptInputs scenario or in error handling along the way.